libraries/Microsoft.Bot.Connector/Authentication/JwtTokenExtractor.cs

```diff
@@ -175,11 +190,23 @@ private async Task<ClaimsPrincipal> ValidateTokenAsync(string jwtToken, string c
                 // below won't run. This is normal.
                 if (!string.IsNullOrEmpty(keyId) && endorsements.TryGetValue(keyId, out var endorsementsForKey))
                 {
+                    // Verify that channelId is included in endorsements
                     var isEndorsed = EndorsementsValidator.Validate(channelId, endorsementsForKey);
+
                     if (!isEndorsed)
                     {
                         throw new UnauthorizedAccessException($"Could not validate endorsement for key: {keyId} with endorsements: {string.Join(",", endorsementsForKey)}");
                     }
+
+                    // Verify that additional endorsements are satisfied. If no additional endorsements are expected, the requirement is satisfied as well
+                    var additionalEndorsementsSatisfied = requiredEndorsements == null ? true : 
+                        requiredEndorsements.All(
```

Comment:
If you remove the null possiblity from the class (and the default is just an empty list), this can simply always run. That makes the logic much easier to follow.